lib/router.js

```diff
@@ -24,19 +24,11 @@ var SwaggerRouter = swaggerRouter.Router;
  *      - {object} operations, Object mapping operationId -> handler
  */
 function ApiScope(init) {
-    if (!init) {
-        init = {};
-    }
-    this.prefixPath = init.prefixPath || '';
-
-    var specRoot = this.specRoot = init.specRoot || {};
-    if (!specRoot.paths) { specRoot.paths = {}; }
-    if (!specRoot.definitions) { specRoot.definitions = {}; }
-    if (!specRoot.securityDefinitions) { specRoot.securityDefinitions = {}; }
-    if (specRoot.basePath === undefined) { specRoot.basePath  = this.prefixPath; }
-
+    init = init || {};
+    this.specRoot = init.specRoot;
```

Comment:
Oh, I meant a different thing actually. Now the code is changed in a way that it must never access specRoot before it have set it up. So, if specRoot is undefined and accessed, that's an error.